htdocs/fourn/class/fournisseur.facture-rec.class.php

```diff
@@ -242,6 +242,7 @@ public function create($user, $facFournId, $notrigger = 0)
 
 		// Clean parameters
 		$this->titre = empty($this->titre) ? '' : $this->titre;
+		$this->ref = $this->{$this->table_ref_field};
```

Comment:
This syntax is controversial and may be deprecated. 
Prefer to use
```
$keyforref = $this->table_ref_field;
$this->ref = $this->$keyforref;

```

Reply:
> This syntax is controversial

I haven't been able to find controversy regarding this particular syntax… I found an RFC for deprecating "${}" inside quotes and the discussion included a debate on the whole concept of variable variable (which would also include `$this->$keyforref`), but the debate was off-topic.

If you have more info on the possible upcoming deprecation of brackets for variable variables, I am interested.